File: OCH_Tao/ABC/B/065.js
const A = (require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n")).map(x => parseInt(x, 10));
const N = A.shift();
let tmp = 1;
let cnt = 0;
let flag = false;
for (let i = 0; i < N; i++) {
  cnt++;
  tmp = A[tmp - 1];
  if (tmp === 2) {
    flag = true;
    break;
  }
}
if (flag) {
  console.log(cnt);
} else {
  console.log(-1);
}

File: OCH_Tao/ABC/B/065.py
N = int(input())
A = []
for i in range(N):
  A.append(int(input()))
tmp = 1
cnt = 0
for i in range(N):
  cnt += 1
  tmp = A[tmp-1]
  if tmp == 2:
    print(cnt)
    break
else:
  print(-1)